controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -0,0 +1,493 @@
+			},
+		},
+	},
+	Status: v1.StorageClusterStatus{
+		FailureDomain: "zone",
+	},
+}
+
+var fakeStorageConsumer = &v1alpha1.StorageConsumer{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-consumer",
+		Namespace: "test-ns",
+	},
+}
+
+var fakeCephFs = &rookCephv1.CephFilesystem{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      fmt.Sprintf("%s-cephfilesystem", fakeStorageCluster.Name),
+		Namespace: fakeStorageCluster.Namespace,
+	},
+	Spec: rookCephv1.FilesystemSpec{
+		DataPools: []rookCephv1.NamedPoolSpec{
+			{
+				PoolSpec: rookCephv1.PoolSpec{
+					DeviceClass: "ssd",
+				},
+			},
+		},
+	},
+}
+
+func createFakeScheme(t *testing.T) *runtime.Scheme {
+
+	scheme, err := v1.SchemeBuilder.Build()
+	if err != nil {
+		assert.Fail(t, "unable to build scheme")
+	}
+
+	err = v1alpha1.AddToScheme(scheme)
+	if err != nil {
+		assert.Fail(t, "failed to add ocsv1alpha1 scheme")
+	}
+
+	err = rookCephv1.AddToScheme(scheme)
+	if err != nil {
+		assert.Fail(t, "failed to add rookCephv1scheme")
+	}
+
+	return scheme
+}
+
+func createFakeReconciler(t *testing.T) StorageClassRequestReconciler {
+	var fakeReconciler StorageClassRequestReconciler
+
+	fakeReconciler.Scheme = createFakeScheme(t)
+	fakeReconciler.log = log.Log.WithName("controller_storagecluster_test")
+	fakeReconciler.OperatorNamespace = "test-ns"
+	fakeReconciler.StorageClassRequest = &v1alpha1.StorageClassRequest{}
+	fakeReconciler.cephResourcesByName = map[string]*v1alpha1.CephResourcesSpec{}
+
+	fakeReconciler.storageConsumer = fakeStorageConsumer
+	fakeReconciler.storageCluster = fakeStorageCluster
+
+	return fakeReconciler
+}
+
+func TestProfileReconcile(t *testing.T) {
+	var err error
+	var caseCounter int
+
+	var primaryTestCases = []struct {
+		label           string
+		scrType         string
+		profileName     string
+		failureExpected bool
+		createObjects   []runtime.Object
+	}{
+		{
+			label:       "Reconcile blockpool StorageClassRequest",
+			scrType:     "blockpool",
+			profileName: fakeStorageProfile.Name,
+		},
+		{
+			label:       "Reconcile sharedfilesystem StorageClassRequest",
+			scrType:     "sharedfilesystem",
+			profileName: fakeStorageProfile.Name,
+		},
+		{
+			label:   "Reconcile blockpool StorageClassRequest with default StorageProfile",
+			scrType: "blockpool",
+		},
+		{
+			label:   "Reconcile sharedfilesystem StorageClassRequest with default StorageProfile",
+			scrType: "sharedfilesystem",
+		},
+		{
+			label:           "Reconcile blockpool StorageClassRequest with invalid StorageProfile",
+			scrType:         "blockpool",
+			profileName:     "nope",
+			failureExpected: true,
+		},
+		{
+			label:           "Reconcile sharedfilesystem StorageClassRequest with invalid StorageProfile",
+			scrType:         "sharedfilesystem",
+			profileName:     "nope",
+			failureExpected: true,
+		},
+	}
+
+	for _, c := range primaryTestCases {
+		caseCounter++
+		caseLabel := fmt.Sprintf("Case %d: %s", caseCounter, c.label)
+		fmt.Println(caseLabel)
```

Comment:
why not subtest this and remove print and counter?
`t.Run(caseLabel, func(t * testing.T) {})`

same for remaining tests.